internal/blueprint/blueprint.go

```diff
@@ -31,6 +35,22 @@ type Group struct {
 	Name string `json:"name" toml:"name"`
 }
 
+// DeepCopy returns a deep copy of the blueprint
+// This uses json.Marshal and Unmarshal which are not very efficient
+func (b *Blueprint) DeepCopy() (Blueprint, error) {
```

Comment:
This does the job, but it feels a bit odd to me. Maybe consider (for a follow-up) to either do a proper deep-copy implementation for all the fields (and see if anything can be turned from pointers to embedded members to make it simpler), or simply construct a new Blueprint based on the old and the additional information.

Reply:
Yeah, it feels a bit dirty to me as well, but the alternative is a huge pile of code for all the fields, and given the number of pointers involved I guarantee that's going to have bugs.
I was hoping someone would suggest a well-known deepcopy library -- but this will work for now :)